platform/darwin/src/MGLTileSource.mm

```diff
@@ -129,5 +130,22 @@ - (NSString *)attributionHTMLString {
         }
     }
 
+    if (NSNumber *demEncodingNumber = options[MGLTileSourceOptionDEMEncoding]) {
+        if (![demEncodingNumber isKindOfClass:[NSValue class]]) {
+            [NSException raise:NSInvalidArgumentException
+                        format:@"MGLTileSourceOptionDEMEncoding must be set to an NSValue or NSNumber."];
+        }
+        MGLDEMEncoding demEncoding;
+        [demEncodingNumber getValue:&demEncoding];
+        switch (demEncoding) {
+            case MGLDEMEncodingMapbox:
+                tileSet.encoding = mbgl::Tileset::DEMEncoding::Mapbox;
+                break;
+            case MGLDEMEncodingTerrarium:
+                tileSet.encoding = mbgl::Tileset::DEMEncoding::Terrarium;
```

Comment:
If `encoding` is supported by `TileSet` no matter which source type the `TileSet` is used with, then this implementation is fine and the second and third steps in https://github.com/mapbox/mapbox-gl-native/issues/11224#issuecomment-366844422 are unnecessary.